airflow/operators/python.py

```diff
@@ -145,6 +145,69 @@ def execute_callable(self):
         return self.python_callable(*self.op_args, **self.op_kwargs)
 
 
+class PythonFunctionalOperator(PythonOperator):
+    """
+    Allows a workflow to "branch" or follow a path following the execution
+    of this task.
+
+    It derives the PythonOperator and expects a Python function that returns
+    a single task_id or list of task_ids to follow. The task_id(s) returned
+    should point to a task directly downstream from {self}. All other "branches"
+    or directly downstream tasks are marked with a state of ``skipped`` so that
+    these paths can't move forward. The ``skipped`` states are propagated
+    downstream to allow for the DAG state to fill up and the DAG run's state
+    to be inferred.
+    """
+
+    @apply_defaults
+    def __init__(
+        self,
+        multiple_outputs: bool = False,
+        *args,
+        **kwargs
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.multiple_outputs = multiple_outputs
+        
+
+    def __call__(self, *args, **kwargs):
+        self.op_args = args
+        self.op_kwargs = kwargs
+        # To add when we get XCom merged
+        # return XComArg(self)
+
+    def alias(self, task_id: str):
+        return PythonFunctionalOperator(
```

Comment:
if we create a new operator, it probably should copy all `kwargs` from the current one )
we can add copy function via standard approach or some hacky one via .copy and params modification.

Reply:
Took a stab at this. Maybe its a bit too hacky? Also, need to test this, not sure if it works.

Reply:
@turbaszek what do you think?
maybe we can just use   `def task_id(task_id)`, so the code will look like 
 `some_op.task_id("new_task_id")`. (clear for airflow users, they are used to task_id)

Reply:
If we go this way I am in favor of `.task_id("xxx")`. However I would still consider autogeneration of task_ids

Reply:
It looks like we have to do both.

Reply:
I have an idea to generate it by default (and allow custom task_id). Let me write that 😄

Reply:
Check `__init__` method.